applications/xcm-tools.md

```diff
@@ -0,0 +1,162 @@
+# XCM Tools
+
+* **Team Name:** [gmajor](https://github.com/gmajor-encrypt)
+* **Payment Address:** 0xC3094f0ddce699a1Ad9Ef2621DF68Cd297a4c44F(USDC)
+* **[Level](https://github.com/w3f/Grants-Program/tree/master#level_slider-levels):** 2
+
+## Project Overview :page_facing_up:
+
+RFPs [https://github.com/w3f/Grants-Program/blob/master/docs/RFPs/Open/xcm-tool.md](https://github.com/w3f/Grants-Program/blob/master/docs/RFPs/Open/xcm-tool.md)
+
+### Overview
+
+XCM (Cross-Consensus Message) is a cross-chain message between parachain and relaychain because the xcm format (
+Instruction, MultiAssets, and MultiLocation) are too complex, and no good tool exists. It has yet to be widely used.
+
+XCM Tools is a tool and an SDK. This library is written in golang. It provides the following functions: sending xcm
+messages, parsing xcm message instructions, getting the execution result of the execution after sending xcm
+
+### Project Details
+
+We only have two libs to choose from, and some functions still need to be included.
+
+[Go-Substrate-RPC-Client](https://github.com/centrifuge/go-substrate-rpc-rient)
+Centrifuge's open source and supports transaction and chain status queries. Scale Codec is a static type. Types analysis
+many
+require users definition, and lack of support for Metadata V14 Types
```

Comment:
GSRPC supports Metadata V14, IIRC.

Reply:
As far as I know, currently GSRPC only supports parsing metadata v14, but does not support dynamic type registration, they are still under development (https://github.com/w3f/Grants-Program/blob/master/applications/centrifuge-gsrpc- v2.md).